lib/types/object/index.js

```diff
@@ -366,6 +367,18 @@ internals.Object = class extends Any {
         });
     }
 
+    class() {
+
```

Comment:
Should I add a check right here to see if `this._flags.func === true`?

Reply:
The way that this is setup right now the flag `this._flags.func` could have not been set. The reason why being that a user could do `Joi.object().class()` or `Joi.func().class()`.

Additionally the flag `this._flags.func` just ensures that we have set the flag that it should be a function, it doesn't validate that the input coming into `isClass` is actually a function. I would personally use `typeof func` in the `isClass` method.